From hjk@linutronix.de  Mon Oct 27 11:23:22 2008
From: Arjan van de Ven <arjan@linux.intel.com>
Date: Tue, 21 Oct 2008 11:17:51 +0200
Subject: UIO: use pci_ioremap_bar() in drivers/uio
To: Arjan van de Ven <arjan@infradead.org>
Cc: hjk@linutronix.de, tglx@tglx.de, Greg KH <gregkh@suse.de>, LKML <linux-kernel@vger.kernel.org>
Message-ID: <20081021091750.GA2966@local>
Content-Disposition: inline

From: Arjan van de Ven <arjan@linux.intel.com>

Use the newly introduced pci_ioremap_bar() function in drivers/uio.
pci_ioremap_bar() just takes a pci device and a bar number, with the goal
of making it really hard to get wrong, while also having a central place
to stick sanity checks.

Signed-off-by: Arjan van de Ven <arjan@linux.intel.com>
Signed-off-by: Hans J. Koch <hjk@linutronix.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/uio/uio_cif.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/drivers/uio/uio_cif.c
+++ b/drivers/uio/uio_cif.c
@@ -57,8 +57,7 @@ static int __devinit hilscher_pci_probe(
 	info->mem[0].addr = pci_resource_start(dev, 0);
 	if (!info->mem[0].addr)
 		goto out_release;
-	info->mem[0].internal_addr = ioremap(pci_resource_start(dev, 0),
-					     pci_resource_len(dev, 0));
+	info->mem[0].internal_addr = pci_ioremap_bar(dev, 0);
 	if (!info->mem[0].internal_addr)
 		goto out_release;
 
